src/external_acl.cc

```diff
@@ -999,9 +1001,11 @@ externalAclHandleReply(void *data, const Helper::Reply &reply)
         entry = external_acl_cache_add(state->def, state->key, entryData);
 
     do {
-        void *cbdata;
-        if (state->callback && cbdataReferenceValidDone(state->callback_data, &cbdata))
-            state->callback(cbdata, entry);
+        CallBack(state->codeContext, [&] {
+            void *cbdata = nullptr;
+            if (state->callback && cbdataReferenceValidDone(state->callback_data, &cbdata))
+                state->callback(cbdata, entry);
+        });
```

Comment:
We should not switch to the callback context if that callback is no longer valid (or was not even set because this was an in-the-background "internal" cache-refreshing lookup). AFAICT, the code context switch should be _inside_ the official `if` statement.